tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -62,7 +71,22 @@ static int msg_handler(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 	prev_num = num;
 
 	/* Ensure that no message is received while Mesh is suspended or disabled. */
-	ASSERT_FALSE_MSG(suspended, "Received publication while Mesh is suspended.");
+	if (dut_status == DUT_SUSPENDED) {
+		if (num == (NUM_PUB + 1)) {
+			/* A publication may be sent just before DUT is suspended. Do nothing. */
+		} else if (SUSPEND_DURATION*1000 <= k_uptime_delta(&uptime)) {
+			dut_status = DUT_RUNNING;
+			LOG_DBG("Suspend duration passed. Setting status to %d.", dut_status);
+		} else {
+			FAIL("Received publication while Mesh is suspended.");
+		}
+	}
+
+	if (num == NUM_PUB) {
+		dut_status = DUT_SUSPENDED;
+		LOG_DBG("Expecting number of pubs received. Setting status to %d.", dut_status);
+		uptime = k_uptime_get();
```

Comment:
uptime has been already updated before.

Reply:
This is the only call to `k_uptime_get`, and the uptime set here is used to check that the suspend duration has passed when the DUT starts sending messages again.

Reply:
This is the first time uptime is set during the test. The uptime is updated by `k_uptime_delta`, but at this point the status will either get updated to `DUT_RUNNING` (suspend duration passed) or the test will fail due to receiving a message during suspension.